hail/python/hailtop/batch/docker.py

```diff
@@ -78,15 +81,15 @@ def build_python_image(fullname: str,
     python3 -m pip check
 ''')
 
-            sync_check_shell_output(f'docker build -t {fullname} {docker_path}')
+            sync_check_shell(f'docker build -t {fullname} {docker_path}', inherit_std_out_err=show_docker_output)
             print(f'finished building image {fullname}')
         else:
-            sync_check_shell_output(f'docker pull {base_image}')
-            sync_check_shell_output(f'docker tag {base_image} {fullname}')
+            sync_check_shell(f'docker pull {base_image}', inherit_std_out_err=show_docker_output)
+            sync_check_shell(f'docker tag {base_image} {fullname}', inherit_std_out_err=show_docker_output)
             print(f'finished pulling image {fullname}')
 
         if '/' in fullname:
-            sync_check_shell_output(f'docker push {fullname}')
+            sync_check_shell(f'docker push {fullname}', inherit_std_out_err=show_docker_output)
```

Comment:
These really should not be check shell, or at least we need to shell quote the paths and full name.

---

That suggests these should really be `sync_check_exec`, which does not yet exist.

I feel like we're building up a parallel process API that diverges from the [`subprocess.run` API](https://docs.python.org/3/library/subprocess.html?highlight=subprocess).

If we need to change our asyncio process API, I think we should prefer starting an asyncio copy of `subprocess.run`. In particular, `inherit_std_out_err` is known in `subprocess.run` as `capture_output=False`.

---

That said, this is synchronous code, so maybe we should just use `subprocess.run`? Unfortunately subprocess.run does not include the stdout and stderr in the CalledProcessError:
```
In [33]: subprocess.run(['/bin/bash', '-c', 'echo hello ; echo >&2 \'ERROR!\' ; false'], capture_output=True, check=True) 
    ...:                                                                                                                                          
Traceback (most recent call last):
  File "<ipython-input-33-3dac52dfa019>", line 1, in <module>
    subprocess.run(['/bin/bash', '-c', 'echo hello ; echo >&2 \'ERROR!\' ; false'], capture_output=True, check=True)
  File "/Users/dking/miniconda3/lib/python3.7/subprocess.py", line 512, in run
    output=stdout, stderr=stderr)
CalledProcessError: Command '['/bin/bash', '-c', "echo hello ; echo >&2 'ERROR!' ; false"]' returned non-zero exit status 1.
```

Unfortunately the printed form of CalledProcessError does not include the stdout and stderr. We might want to write our own shim around `subprocess.run` which just raises a Hail CalledProcessError with includes the stdout and stderr (which are available on subprocess.CalledProcessError as `stdout` and `stderr`).